service/matching/reachability.go

```diff
@@ -36,16 +36,23 @@ import (
 
 	"go.temporal.io/server/api/clock/v1"
 	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/cache"
 	hlc "go.temporal.io/server/common/clock/hybrid_logical_clock"
+	"go.temporal.io/server/common/metrics"
 	"go.temporal.io/server/common/namespace"
 	"go.temporal.io/server/common/persistence/visibility/manager"
 	"go.temporal.io/server/common/searchattribute"
 	"go.temporal.io/server/common/util"
 	"go.temporal.io/server/common/worker_versioning"
 )
 
+const (
+	reachabilityCacheMaxSize = 100
```

Comment:
I think we should increase the size (maybe 10K). Cached items should be small so no concern of high cache usage. Also it's worth making this a config as well.